### ISSUE TITLE
Layout fine-tuning for the 5.5.0 release and the future of React-mde

### ISSUE THREAD

**Issue by andrerpena** [andrerpena/react-mde#103]
@Nes-si , I'm opening this issue just so you know what I'm thinking and working on, and so that we can discuss the future of `react-mde`.

I think `react-mde` is almost prod ready, and you played a big role on that, except for these things, that I'm working on right now:

- [ ] The UI is still a bit clumsy, it doesn't fill in place quite yet.
- [ ] `Undo` works but `Redo` after commands are applied is still a bit messy and unpredictable.

I'm going to release 5.5 when the first is ready, and 5.6 after the second is ready. After that, my focus will be on:

- [ ] Support for pasting files like images
- [ ] Support for mentioning.

Now back to the layout, which is the focus now. I personally think that when you chose a `vertical layout`, you don't want to toggle the preview. The same applies for `horizontal`. The reason being that, if you don't want the users to see the preview, just use the `noPreview` layout. If you want toggling, just use the  `tabbed` layout. This is how I see it.

However, in respect for your work. I want to keep the toggling. Basically we have 2 options: 

1. Make toggling optional by passing it in the `layoutOptions` prop. Whatever you pass to `layoutOptions` is passed to the layout component and it can react to it as it sees fit.
2. Make the toggable layouts separate. I mean, you would have `horizontal` and `horizontal-toggable`. The same for `vertical`.

Both has pros and cons. After thinking about it, I prefer the first one, because, even though it will make the layout components a bit bloated because it will have to treat the props, it is still better because we will have less layouts to maintain and specially less SASS files. Assuming each layout has its own SASS, having 2 more layouts will add more complication than having the option to toggle the toggle buttons. I'll make the toggle buttons invisible by default.

@Nes-si , let me know what you think.

**Comment by andrerpena** [comment]
Also, @Nes-si , I'd like to make the `tabbed` layout as similar as possible to GitHub so, if you don't mind, I'm moving the toolbar buttons to the right.

**Comment by andrerpena** [comment]
@Nes-si , I'm trying to make a final version for 5.5, especially when it comes to UI/UX, and here is where I came to:  We do not have resources (time and contributors) to make a feature-rich, stable and beautifully designed product with too many options. I'd rather have a simpler editor, with less options but meeting what we've proposed to do than a product with more features that is somewhat half-baked.

For that reason I decided to drop support for toggling the preview and editors for the vertical and horizontal layouts. Toggling will only be supported for the tabbed layout, and I'll tweak it to look a bit more like GitHub.

I hope you understand and thanks again for your contributions ❤️
